dashboard/test/models/user_test.rb

```diff
@@ -1046,6 +1046,61 @@ def complete_script_for_user(user, script, completed_date = Time.now)
     assert @student.can_edit_email?
   end
 
+  test 'teacher_managed_account? is false for teacher' do
+    refute @teacher.teacher_managed_account?
+  end
+
+  test 'teacher_managed_account? is false for normal student account with hashed email and password' do
+    refute @student.teacher_managed_account?
+  end
+
+  test 'teacher_managed_account? is false for student account in section with oauth connection' do
+    student_with_oauth = create(:student, encrypted_password: nil, provider: 'facebook', uid: '1111111')
```

Comment:
TODO(asher): Create an oauth factory as a follow-up to this PR, making use of it here.